be/src/common/config.h

```diff
@@ -849,6 +849,8 @@ CONF_Int32(connector_io_tasks_slow_io_latency_ms, "50");
 CONF_mDouble(scan_use_query_mem_ratio, "0.25");
 CONF_Double(connector_scan_use_query_mem_ratio, "0.3");
 
+// hdfs client socket timeout, default value is 60000ms
+CONF_Int32(hdfs_client_socket_timeout, "60000");
```

Comment:
is there a better way to do this?

There are two obvious cons for this solution.
1) there are too many hdfs client options, this will open the window that more and more hdfs options will be added to be.conf
2) it will be error-prone in be.conf side for consistent management across all be/cn nodes.
3) what if user's core-site.xml contains the socket timeout settings, this change will not respect user's core-site.xml settings.

Reply:
and 4th, hardcode configuration globally if multiple hdfs clusters are used.
5th, more and more inconsistent user experience between external catalog access and shared-data builtin table with hdfs as remote storage.

Reply:
@kevincai Good point. I agree with you. However, I think we can file another PR to make it configure dynamically since it requires us to init hdfs client dynamically